src/config.c

```diff
@@ -20,102 +21,20 @@
-#define READ_INTEGER(opt, default_value)                                       \
-    READ_PRIMITIVE(json_object_get_int, opt, default_value)
-#define READ_FLOAT(opt, default_value)                                         \
-    READ_PRIMITIVE(json_object_get_double, opt, default_value)
-
-#define READ_ENUM(opt, default_value, enum_map)                                \
-    do {                                                                       \
-        g_Config.opt = Config_ReadEnum(                                        \
-            root_obj, Config_ProcessKey(QUOTE(opt)), default_value, enum_map); \
-    } while (0)
-
-#define WRITE_PRIMITIVE(func, opt)                                             \
-    do {                                                                       \
-        func(root_obj, Config_ProcessKey(QUOTE(opt)), g_Config.opt);           \
-    } while (0)
-#define WRITE_BOOL(opt) WRITE_PRIMITIVE(json_object_append_bool, opt)
-#define WRITE_INTEGER(opt) WRITE_PRIMITIVE(json_object_append_int, opt)
-#define WRITE_FLOAT(opt) WRITE_PRIMITIVE(json_object_append_double, opt)
-#define WRITE_ENUM(opt, enum_map)                                              \
-    do {                                                                       \
-        Config_WriteEnum(                                                      \
-            root_obj, Config_ProcessKey(QUOTE(opt)), g_Config.opt, enum_map);  \
-    } while (0)
-
 CONFIG g_Config = { 0 };
 
 static const char *m_TR1XGlobalSettingsPath = "cfg/TR1X.json5";
 
-typedef struct ENUM_MAP {
-    const char *text;
-    int value;
-} ENUM_MAP;
-
-const ENUM_MAP m_UIStyles[] = {
-    { "ps1", UI_STYLE_PS1 },
-    { "pc", UI_STYLE_PC },
-    { NULL, -1 },
-};
-
-const ENUM_MAP m_BarShowingModes[] = {
-    { "default", BSM_DEFAULT },
-    { "flashing-or-default", BSM_FLASHING_OR_DEFAULT },
-    { "flashing-only", BSM_FLASHING_ONLY },
-    { "always", BSM_ALWAYS },
-    { "never", BSM_NEVER },
-    { "ps1", BSM_PS1 },
-    { NULL, -1 },
-};
-
-const ENUM_MAP m_BarLocations[] = {
-    { "top-left", BL_TOP_LEFT },
-    { "top-center", BL_TOP_CENTER },
-    { "top-right", BL_TOP_RIGHT },
-    { "bottom-left", BL_BOTTOM_LEFT },
-    { "bottom-center", BL_BOTTOM_CENTER },
-    { "bottom-right", BL_BOTTOM_RIGHT },
-    { NULL, -1 },
-};
-
-const ENUM_MAP m_BarColors[] = {
-    { "gold", BC_GOLD },   { "blue", BC_BLUE },     { "grey", BC_GREY },
-    { "red", BC_RED },     { "silver", BC_SILVER }, { "green", BC_GREEN },
-    { "gold2", BC_GOLD2 }, { "blue2", BC_BLUE2 },   { "pink", BC_PINK },
-    { NULL, -1 },
-};
-
-const ENUM_MAP m_ScreenshotFormats[] = {
-    { "jpg", SCREENSHOT_FORMAT_JPEG },
-    { "jpeg", SCREENSHOT_FORMAT_JPEG },
-    { "png", SCREENSHOT_FORMAT_PNG },
-    { NULL, -1 },
-};
-
-static const char *Config_ProcessKey(const char *key);
 static int Config_ReadEnum(
     struct json_object_s *obj, const char *name, int8_t default_value,
```

Comment:
Should we change the `int8_t` here to `int32_t` to match what's being passed in?

Reply:
I changed it to an `int` because ints are commonly used for enums and we also use ints in other parts of the code. However, it's important to note that ints are not always guaranteed to be 32-bit. In order to ensure this guarantee, I'd need to change `g_Config` members to be of type `int` instead of the relevant enums. Nonetheless, I don't think this is ever going to be relevant to us.